packages/eslint-plugin-js/rules/array-bracket-newline/array-bracket-newline.js

```diff
@@ -3,256 +3,238 @@
  * @author Jan Peer Stöcklmair <https://github.com/JPeer264>
  */
 
-'use strict'
-
-const astUtils = require('../../utils/ast-utils')
+import astUtils from '../../utils/ast-utils.js'
 
 // ------------------------------------------------------------------------------
 // Rule Definition
 // ------------------------------------------------------------------------------
 
 /** @type {import('eslint').Rule.RuleModule} */
-module.exports = {
-  meta: {
-    type: 'layout',
+export const meta = {
```

Comment:
I don't think it's necessary to export `meta` and `create`.
I think it is enough to just export the rules to `default`. `ts` rules only have `export default`. What do you think?
https://github.com/eslint-stylistic/eslint-stylistic/blob/0a77e190c8f74145118c3ad7125637de05c63adb/packages/eslint-plugin-ts/rules/block-spacing/block-spacing.ts#L16

Reply:
I could also just locally scope `meta` and `create` since it's being exported anyway as `default` on line 237. And if you check the tester file, it anyways consumes the module as `rule` on line 11, so I just kept the behavior from CJS over to ESM. 

Or do you mean to take the route the the TS implementation took with utils for creating rules and also the meta information as well like such: 
```
/* Check RuleCreator.d.ts from @typescript-eslint/utils */
export default createRule({
  // New information explicitly defined using the new method
  name: 'array-bracket-newline',
  meta: {
    // Same meta infomration from previous JS file
    type: 'layout',
    docs: {
      description: 'Enforce linebreaks after opening and before closing array brackets',
      recommended: false,
      url: 'https://eslint.style/rules/js/array-bracket-newline',
    },
    fixable: 'whitespace',
    // Note baseRule received from getESLintCoreRule in ts utils
    hasSuggestions: baseRule.meta.hasSuggestions,
    schema: baseRule.meta.schema,
    messages: baseRule.meta.messages,
  },

  create() {
    /* Implementation */
    return {
      ArrayPattern: check,
      ArrayExpression: check,
    }
  },
})
```

Reply:
Only downside of doing such approach is then people would not be able to selectively import the `meta` information like `import { meta } from './array-bracket-newline'` if they are looking to extend certain parts of the code in the future.

Reply:
I think you can reduce the difference by `export default { meta: {...}, create () {...}}`.
Now that indentation has changed significantly, I think reviewers need to spend a lot of time.
If you want to do another refactoring beyond making it ESM, it may be better to make it a separate PR.

Reply:
Latest commit reflect the points above and maintained as little changed code as possible for easier review :)